src/algebra/char_p/basic.lean

```diff
@@ -17,7 +14,68 @@ import ring_theory.nilpotent
 
 universes u v
 
-variables (R : Type u)
+open_locale big_operators
+
+variables {R : Type*}
+
+namespace commute
+variables [semiring R] {p : ℕ} {x y : R}
+
+protected lemma add_pow_prime_pow_eq (hp : p.prime) (h : commute x y) (n : ℕ) :
+  (x + y) ^ p ^ n = x ^ p ^ n + y ^ p ^ n +
+    p * ∑ k in finset.range (p ^ n - 1),
+      x ^ (k + 1) * y ^ (p ^ n - (k + 1)) * ↑((p ^ n).choose (k + 1) / p) :=
```

Comment:
This (and the rest) would arguably be nicer at least to read as
```suggestion
    p * ∑ k in finset.Ioo 0 (p ^ n),
      x ^ k * y ^ (p ^ n - k) * ↑((p ^ n).choose k / p) :=
```
but I don't know if that's more annoying to prove things about.

bors d+

Reply:
#18533 should make this straighforward

Reply:
Are you happy with the lines I needed to add?

Reply:
I'd rather go through #18533 which I think provides a more direct proof

Reply:
What lemma do you want to use from there? I don't see it.